src/Compute/v2/Params.php

```diff
@@ -114,9 +114,9 @@ public function imageId(): array
     {
         return [
             'type'        => self::STRING_TYPE,
-            'required'    => true,
+            'required'    => false,
             'sentAs'      => 'imageRef',
-            'description' => 'The unique ID of the image that this server will be based on',
+            'description' => 'The UUID of the image to use for your server instance. This is not required in case of boot from volume. In all other cases it is required and must be a valid UUID',
```

Comment:
Change [this line](https://github.com/nexcess/openstack/blob/master/src/Compute/v2/Api.php#L193) to:

```
$this->notRequired($this->params->imageId())
```

and add logic to [this method](https://github.com/nexcess/openstack/blob/master/src/Compute/v2/Models/Server.php#L107) which ensures either `imageId` or BFV param is provided